File: StyleGuide-iOS/StyleGuide/Colors/UIColor+Allinson.m
//
//  UIColor+Allinson.m
//  StyleGuide-iOS
//
//  Created by Chris Allinson on 2024-05-20.
//

#import "UIColor+Allinson.h"
#import "GeneratedAssetSymbols.h"

@implementation UIColor (Allinson)

+ (UIColor *)Allinson_header_brown {
    UIColor *c = [UIColor colorNamed:ACColorNameAllinsonHeaderBrown];
    if (c) {
        return c;
    } else {
        if ([[[UIScreen mainScreen] traitCollection] userInterfaceStyle] == UIUserInterfaceStyleDark) {
            return [[UIColor alloc] initWithRed:(42.0/255.0) green:(35.0/255.0) blue:(22.0/255.0) alpha:1.0];
        } else {
            return [[UIColor alloc] initWithRed:(214.0/255.0) green:(202.0/255.0) blue:(149.0/255.0) alpha:1.0];
        }
    }
}

+ (UIColor *)Allinson_body_brown {
    UIColor *c = [UIColor colorNamed:ACColorNameAllinsonBodyBrown];
    if (c) {
        return c;
    } else {
        if ([[[UIScreen mainScreen] traitCollection] userInterfaceStyle] == UIUserInterfaceStyleDark) {
            return [[UIColor alloc] initWithRed:(50.0/255.0) green:(43.0/255.0) blue:(30.0/255.0) alpha:1.0];
        } else {
            return [[UIColor alloc] initWithRed:(222.0/255.0) green:(210.0/255.0) blue:(158.0/255.0) alpha:1.0];
        }
    }
}


+ (UIColor *)Allinson_crushed_brown {
    UIColor *c = [UIColor colorNamed:ACColorNameAllinsonCrushedBrown];
    if (c) {
        return c;
    } else {
        if ([[[UIScreen mainScreen] traitCollection] userInterfaceStyle] == UIUserInterfaceStyleDark) {
            return [[UIColor alloc] initWithRed:(214.0/255.0) green:(202.0/255.0) blue:(149.0/255.0) alpha:1.0];
        } else {
            return [[UIColor alloc] initWithRed:(104.0/255.0) green:(86.0/255.0) blue:(66.0/255.0) alpha:1.0];
        }
    }
}

+ (UIColor *)Allinson_crushed_brown__disabled {
    UIColor *c = [UIColor colorNamed:ACColorNameAllinsonCrushedBrownDisabled];
    if (c) {
        return c;
    } else {
        if ([[[UIScreen mainScreen] traitCollection] userInterfaceStyle] == UIUserInterfaceStyleDark) {
            return [[UIColor alloc] initWithRed:(214.0/255.0) green:(202.0/255.0) blue:(149.0/255.0) alpha:0.3];
        } else {
            return [[UIColor alloc] initWithRed:(104.0/255.0) green:(86.0/255.0) blue:(66.0/255.0) alpha:0.3];
        }
    }
}


+ (UIColor *)Allinson_copy_brown {
    UIColor *c = [UIColor colorNamed:ACColorNameAllinsonCopyBrown];
    if (c) {
        return c;
    } else {
        if ([[[UIScreen mainScreen] traitCollection] userInterfaceStyle] == UIUserInterfaceStyleDark) {
            return [[UIColor alloc] initWithRed:(214.0/255.0) green:(202.0/255.0) blue:(149.0/255.0) alpha:1.0];
        } else {
            return [[UIColor alloc] initWithRed:(50.0/255.0) green:(43.0/255.0) blue:(30.0/255.0) alpha:1.0];
        }
    }
}

+ (UIColor *)Allinson_copy_brown__disabled {
    UIColor *c = [UIColor colorNamed:ACColorNameAllinsonCopyBrownDisabled];
    if (c) {
        return c;
    } else {
        if ([[[UIScreen mainScreen] traitCollection] userInterfaceStyle] == UIUserInterfaceStyleDark) {
            return [[UIColor alloc] initWithRed:(214.0/255.0) green:(202.0/255.0) blue:(149.0/255.0) alpha:0.3];
        } else {
            return [[UIColor alloc] initWithRed:(50.0/255.0) green:(43.0/255.0) blue:(30.0/255.0) alpha:0.3];
        }
    }
}


+ (UIColor *)Allinson_orange {
    UIColor *c = [UIColor colorNamed:ACColorNameAllinsonOrange];
    if (c) {
        return c;
    } else {
        if ([[[UIScreen mainScreen] traitCollection] userInterfaceStyle] == UIUserInterfaceStyleDark) {
            return [[UIColor alloc] initWithRed:(255.0/255.0) green:(172.0/255.0) blue:(58.0/255.0) alpha:1.0];
        } else {
            return [[UIColor alloc] initWithRed:(204.0/255.0) green:(102.0/255.0) blue:(0.0/255.0) alpha:1.0];
        }
    }
}


+ (UIColor *)Allinson_link_blue {
    UIColor *c = [UIColor colorNamed:ACColorNameAllinsonLinkBlue];
    if (c) {
        return c;
    } else {
        if ([[[UIScreen mainScreen] traitCollection] userInterfaceStyle] == UIUserInterfaceStyleDark) {
            return [[UIColor alloc] initWithRed:(115.0/255.0) green:(215.0/255.0) blue:(255.0/255.0) alpha:1.0];
        } else {
            return [[UIColor alloc] initWithRed:(0.0/255.0) green:(0.0/255.0) blue:(255.0/255.0) alpha:1.0];
        }
    }
}


+ (UIColor *)Allinson_red {
    UIColor *c = [UIColor colorNamed:ACColorNameAllinsonRed];
    if (c) {
        return c;
    } else {
        if ([[[UIScreen mainScreen] traitCollection] userInterfaceStyle] == UIUserInterfaceStyleDark) {
            return [[UIColor alloc] initWithRed:(255.0/255.0) green:(41.0/255.0) blue:(41.0/255.0) alpha:1.0];
        } else {
            return [[UIColor alloc] initWithRed:(153.0/255.0) green:(41.0/255.0) blue:(41.0/255.0) alpha:1.0];
        }
    }
}

+ (UIColor *)Allinson_red__darkened {
    UIColor *c = [UIColor colorNamed:ACColorNameAllinsonRedDarkend];
    if (c) {
        return c;
    } else {
        if ([[[UIScreen mainScreen] traitCollection] userInterfaceStyle] == UIUserInterfaceStyleDark) {
            return [[UIColor alloc] initWithRed:(247.0/255.0) green:(33.0/255.0) blue:(33.0/255.0) alpha:1.0];
        } else {
            return [[UIColor alloc] initWithRed:(122.0/255.0) green:(33.0/255.0) blue:(33.0/255.0) alpha:1.0];
        }
    }
}


+ (UIColor *)Allinson_footer_green {
    UIColor *c = [UIColor colorNamed:ACColorNameAllinsonFooterGreen];
    if (c) {
        return c;
    } else {
        if ([[[UIScreen mainScreen] traitCollection] userInterfaceStyle] == UIUserInterfaceStyleDark) {
            return [[UIColor alloc] initWithRed:(72.0/255.0) green:(78.0/255.0) blue:(55.0/255.0) alpha:1.0];
        } else {
            return [[UIColor alloc] initWithRed:(183.0/255.0) green:(198.0/255.0) blue:(139.0/255.0) alpha:1.0];
        }
    }
}

+ (UIColor *)Allinson_footer_green__darkened {
    UIColor *c = [UIColor colorNamed:ACColorNameAllinsonFooterGreenDarkened];
    if (c) {
        return c;
    } else {
        if ([[[UIScreen mainScreen] traitCollection] userInterfaceStyle] == UIUserInterfaceStyleDark) {
            return [[UIColor alloc] initWithRed:(64.0/255.0) green:(70.0/255.0) blue:(47.0/255.0) alpha:1.0];
        } else {
            return [[UIColor alloc] initWithRed:(146.0/255.0) green:(158.0/255.0) blue:(111.0/255.0) alpha:1.0];
        }
    }
}

@end

File: StyleGuide-iOS/StyleGuide/Colors/UIColor+Allinson.swift
//
//  UIColor+Allinson.swift
//  Allinson iOS StyleGuide
//
//  Created by Chris Allinson on 2024-05-19.
//  Copyright © 2024 Chris Allinson. All rights reserved.
//

import UIKit

public extension UIColor {
    
    struct Allinson {
        
        // MARK: browns
        
        public static var header_brown: UIColor {
            guard let tempColor = UIColor(named: "Allinson_header_brown") else {
                #if os(iOS)
                if UIScreen.main.traitCollection.userInterfaceStyle == .dark {
                    return UIColor(red: 42.0/255.0, green: 35.0/255.0, blue: 22.0/255.0, alpha: 1.0)
                } else {
                    return UIColor(red: 214.0/255.0, green: 202.0/255.0, blue: 149.0/255.0, alpha: 1.0)
                }
                #else
                return UIColor(red: 214.0/255.0, green: 202.0/255.0, blue: 149.0/255.0, alpha: 1.0)
                #endif
            }
            return tempColor
        }
        
        public static var body_brown: UIColor {
            guard let tempColor = UIColor(named: "Allinson_body_brown") else {
                #if os(iOS)
                if UIScreen.main.traitCollection.userInterfaceStyle == .dark {
                    return UIColor(red: 50.0/255.0, green: 43.0/255.0, blue: 30.0/255.0, alpha: 1.0)
                } else {
                    return UIColor(red: 222.0/255.0, green: 210.0/255.0, blue: 158.0/255.0, alpha: 1.0)
                }
                #else
                return UIColor(red: 222.0/255.0, green: 210.0/255.0, blue: 158.0/255.0, alpha: 1.0)
                #endif
            }
            return tempColor
        }
        
        public static var crushed_brown: UIColor {
            guard let tempColor = UIColor(named: "Allinson_crushed_brown") else {
                #if os(iOS)
                if UIScreen.main.traitCollection.userInterfaceStyle == .dark {
                    return UIColor(red: 214.0/255.0, green: 202.0/255.0, blue: 149.0/255.0, alpha: 1.0)
                } else {
                    return UIColor(red: 104.0/255.0, green: 86.0/255.0, blue: 66.0/255.0, alpha: 1.0)
                }
                #else
                return UIColor(red: 104.0/255.0, green: 86.0/255.0, blue: 66.0/255.0, alpha: 1.0)
                #endif
            }
            return tempColor
        }
        
        public static var crushed_brown__disabled: UIColor {
            guard let tempColor = UIColor(named: "Allinson_crushed_brown__disabled") else {
                #if os(iOS)
                if UIScreen.main.traitCollection.userInterfaceStyle == .dark {
                    return UIColor(red: 214.0/255.0, green: 202.0/255.0, blue: 149.0/255.0, alpha: 0.3)
                } else {
                    return UIColor(red: 104.0/255.0, green: 86.0/255.0, blue: 66.0/255.0, alpha: 0.3)
                }
                #else
                return UIColor(red: 104.0/255.0, green: 86.0/255.0, blue: 66.0/255.0, alpha: 0.3)
                #endif
            }
            return tempColor
        }
        
        public static var copy_brown: UIColor {
            guard let tempColor = UIColor(named: "Allinson_copy_brown") else {
                #if os(iOS)
                if UIScreen.main.traitCollection.userInterfaceStyle == .dark {
                    return UIColor(red: 214.0/255.0, green: 202.0/255.0, blue: 149.0/255.0, alpha: 1.0)
                } else {
                    return UIColor(red: 50.0/255.0, green: 43.0/255.0, blue: 30.0/255.0, alpha: 1.0)
                }
                #else
                return UIColor(red: 50.0/255.0, green: 43.0/255.0, blue: 30.0/255.0, alpha: 1.0)
                #endif
            }
            return tempColor
        }
        
        public static var copy_brown__disabled: UIColor {
            guard let tempColor = UIColor(named: "Allinson_copy_brown__disabled") else {
                #if os(iOS)
                if UIScreen.main.traitCollection.userInterfaceStyle == .dark {
                    return UIColor(red: 214.0/255.0, green: 202.0/255.0, blue: 149.0/255.0, alpha: 0.3)
                } else {
                    return UIColor(red: 50.0/255.0, green: 43.0/255.0, blue: 30.0/255.0, alpha: 0.3)
                }
                #else
                return UIColor(red: 50.0/255.0, green: 43.0/255.0, blue: 30.0/255.0, alpha: 0.3)
                #endif
            }
            return tempColor
        }
        
        // MARK: oranges
        
        public static var orange: UIColor {
            guard let tempColor = UIColor(named: "Allinson_orange") else {
                #if os(iOS)
                if UIScreen.main.traitCollection.userInterfaceStyle == .dark {
                    return UIColor(red: 255.0/255.0, green: 172.0/255.0, blue: 58.0/255.0, alpha: 1.0)
                } else {
                    return UIColor(red: 204.0/255.0, green: 102.0/255.0, blue: 0.0/255.0, alpha: 1.0)
                }
                #else
                return UIColor(red: 204.0/255.0, green: 102.0/255.0, blue: 0.0/255.0, alpha: 1.0)
                #endif
            }
            return tempColor
        }
        
        // MARK: blues
        
        public static var link_blue: UIColor {
            guard let tempColor = UIColor(named: "Allinson_link_blue") else {
                #if os(iOS)
                if UIScreen.main.traitCollection.userInterfaceStyle == .dark {
                    return UIColor(red: 115.0/255.0, green: 215.0/255.0, blue: 255.0/255.0, alpha: 1.0)
                } else {
                    return UIColor(red: 0.0/255.0, green: 0.0/255.0, blue: 255.0/255.0, alpha: 1.0)
                }
                #else
                return UIColor(red: 0.0/255.0, green: 0.0/255.0, blue: 255.0/255.0, alpha: 1.0)
                #endif
            }
            return tempColor
        }
        
        // MARK: reds
        
        public static var red: UIColor {
            guard let tempColor = UIColor(named: "Allinson_red") else {
                #if os(iOS)
                if UIScreen.main.traitCollection.userInterfaceStyle == .dark {
                    return UIColor(red: 255.0/255.0, green: 41.0/255.0, blue: 41.0/255.0, alpha: 1.0)
                } else {
                    return UIColor(red: 153.0/255.0, green: 41.0/255.0, blue: 41.0/255.0, alpha: 1.0)
                }
                #else
                return UIColor(red: 153.0/255.0, green: 41.0/255.0, blue: 41.0/255.0, alpha: 1.0)
                #endif
            }
            return tempColor
        }
        
        public static var red__darkened: UIColor {
            guard let tempColor = UIColor(named: "Allinson_red__darkened") else {
                #if os(iOS)
                if UIScreen.main.traitCollection.userInterfaceStyle == .dark {
                    return UIColor(red: 247.0/255.0, green: 33.0/255.0, blue: 33.0/255.0, alpha: 1.0)
                } else {
                    return UIColor(red: 122.0/255.0, green: 33.0/255.0, blue: 33.0/255.0, alpha: 1.0)
                }
                #else
                return UIColor(red: 122.0/255.0, green: 33.0/255.0, blue: 33.0/255.0, alpha: 1.0)
                #endif
            }
            return tempColor
        }
        
        // MARK: greens
        
        public static var footer_green: UIColor {
            guard let tempColor = UIColor(named: "Allinson_footer_green") else {
                #if os(iOS)
                if UIScreen.main.traitCollection.userInterfaceStyle == .dark {
                    return UIColor(red: 72.0/255.0, green: 78.0/255.0, blue: 55.0/255.0, alpha: 1.0)
                } else {
                    return UIColor(red: 183.0/255.0, green: 198.0/255.0, blue: 139.0/255.0, alpha: 1.0)
                }
                #else
                return UIColor(red: 183.0/255.0, green: 198.0/255.0, blue: 139.0/255.0, alpha: 1.0)
                #endif
            }
            return tempColor
        }
        
        public static var footer_green__darkened: UIColor {
            guard let tempColor = UIColor(named: "Allinson_footer_green__darkened") else {
                #if os(iOS)
                if UIScreen.main.traitCollection.userInterfaceStyle == .dark {
                    return UIColor(red: 64.0/255.0, green: 70.0/255.0, blue: 47.0/255.0, alpha: 1.0)
                } else {
                    return UIColor(red: 146.0/255.0, green: 158.0/255.0, blue: 111.0/255.0, alpha: 1.0)
                }
                #else
                return UIColor(red: 146.0/255.0, green: 158.0/255.0, blue: 111.0/255.0, alpha: 1.0)
                #endif
            }
            return tempColor
        }
    }
}
